src/index.ts

```diff
@@ -5,45 +5,80 @@
 //Imports
 import {getInput, info, setSecret} from '@actions/core';
 import getClient from './apollo';
-import {login, publish, setRegistryUrl} from './wapm';
+import {login, publish, setRegistryUrl, tokenLogin} from './wapm';
 
 const main = async () =>
 {
-  //Get input
-  const username = getInput('username', {
-    required: true
-  });
+      //Get input
+      const username = getInput('username');
+      const password = getInput('password');
+      const userToken = getInput('token');
+      if(userToken) {
+        loginWithToken(userToken)
+      } else {
+        loginWithUsername(username, password)
+      }
+  // loginWithUsername()
 
-  const password = getInput('password', {
-    required: true
-  });
-
-  const directory = getInput('directory') != '' ? getInput('directory') : process.cwd();
-
-  const registry = getInput('registry');
-
-  //Hide credentials from the logs
-  setSecret(username);
-  setSecret(password);
+ 
+  //Exit
+  process.exit();
+};
 
-  //Get the Apollo client
-  if (registry) {
-    await setRegistryUrl(registry);
-  }
-  const client = getClient(registry);
 
-  //Login to WAPM
-  await login(client, username, password);
-  info('Logged in to WAPM.');
+// login with username and password 
+const loginWithUsername = async (username: string, password: string) => {
 
-  //Publish to WAPM
-  await publish(directory);
-  info(`Published the package located in ${directory}.`);
+  
+    const directory = getInput('directory') != '' ? getInput('directory') : process.cwd();
+  
+    const registry = getInput('registry');
+  
+    //Hide credentials from the logs
+    setSecret(username);
+    setSecret(password);
+  
+    //Get the Apollo client
+    if (registry) {
+      await setRegistryUrl(registry);
+    }
+    const client = getClient(registry);
+  
+    //Login to WAPM
+    await login(client, username, password);
+    info('Logged in to WAPM.');
+  
+    //Publish to WAPM
+    await publish(directory);
+    info(`Published the package located in ${directory}.`);
+  
+}
 
-  //Exit
-  process.exit();
-};
+const loginWithToken = async (userToken: string) => {
 
+  
+    const directory = getInput('directory') != '' ? getInput('directory') : process.cwd();
+  
+    const registry = getInput('registry');
+  
+    //Hide credentials from the logs
+    setSecret(userToken);
+  
+    //Get the Apollo client
+    if (registry) {
+      await setRegistryUrl(registry);
+    }
+    const client = getClient(registry);
+  
+    //Login to WAPM
+    await tokenLogin(client, userToken);
+    info('Logged in to WAPM.');
+  
+    //Publish to WAPM
+    await publish(directory);
+    info(`Published the package located in ${directory}.`);
```

Comment:
This seems very wrong, why do we need to publish when logging?
We should move this call to the `main` function